horace_core/sqw/@sqw/resolution_plot.m

```diff
@@ -54,76 +60,91 @@
 %               Default: 'e'
 %
 % Plot options:
-%   'over'  If present, then overplot on an existing resolution function plot
-%           If one doesn't already exist, create a new resolution function plot
+%   'curr'          Overplot on the currently active plot
+%
+%   'name', name    Overplot on the existing figure with the given name,
+%                   figure number or figure handle
+%
+%
+% Output:
+% -------
+%   cov_proj    Covariance matrix for wavevector-energy in projection axes
+%               Array size [4,4,npnt] where npnt is the number of points at
+%              which the resolution function was requested.
+%
+%   cov_spec    Covariance matrix for wavevector-energy in spectrometer axes
+%              i.e. x || ki, z vertically upwards, y perpendicular to z and x.
+%               Array size [4,4,npnt] where npnt is the number of points at
+%              which the resolution function was requested.
 %
-%   'curr'  Overplot on the currently active plot
+%   cov_hkle    Covariance matrix for wavevector-energy in h-k-l-energy
+%               Array size [4,4,npnt] where npnt is the number of points at
+%              which the resolution function was requested.
 %
-%   'name', name    Overplot on the named figure or figure number of an existing plot
+%   kept        Logical column vector with length equal to the number of
+%              points given by input argument xres, where elements =1 for
+%              points in bins with contributiung pixels, and =0 for points
+%              in empty bins or outside the range of the data.
 
 
 % The special case of an sqw object with no pixels, a single header and a single
-% detector will be treated as valid
+% detector will be treated as valid. This will have come from the general
+% purpose resolution function tool called resolution_plot that plots without a
+% dataset.
+
 
 
-% Check input arguments and get defaults where necessary
-% ------------------------------------------------------
 % Check sqw object
+% ----------------
 if ~isscalar(w) || dimensions(w)~=2
```

Comment:
Is `dimensions` `ndims`?

Reply:
They are different. dimensions is a method of sqw that returns the dimensionality of the sqw object rather than ndims for the signal array. A one dimensional sqw object will have signal array size (n,1) (n>1); ndims(w.data.signal) will be returned as 2. In tthis context the sqw dimensionality is needed.